### ISSUE TITLE
Support passing in childContext

### ISSUE THREAD

**Issue by jhollingworth** [QubitProducts/react-test-tree#12]
```js
testTest(<SomeComponent />, {
   childContext: {
      foo: 'bar'
   }
});
```

**Comment by jhollingworth** [comment]
This is actually a bit harder than I originally thought since you need to wrap the component before its created. Some options off the top of my head

**Allow you to pass in a factory function**

Simplest although its a bit of a confusing API. If we went down this route i'd recommend we always recommend you pass in a factory function

```js
testTree(() => <SomeComponent />, {
   childContext: {
      foo: 'bar'
   }
});
```

**Wrapper function**

Simpler API (I think) but more code to write

```js
let testTree = require('react-test-tree');
let { withChildContext } = testTree;

var Component = withChidContext(SomeComponent, {
   childContext: {
      foo: 'bar'
   }
});

testTree(<Component />);
```

**Wrapper component**

Simpler API (I think) but a bit clunky to write

```js
let testTree = require('react-test-tree');
let { Context } = testTree;

var childContext = {
   foo: 'bar'
};

testTree( <Context context={childContext}><InnerComponent /></Context>);
```

**Comment by oliverwoodings** [comment]
Closed via #15
